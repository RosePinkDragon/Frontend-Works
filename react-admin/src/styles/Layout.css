header {
  display: flex;
  justify-content: center;
  width: 100%;
  align-items: center;
  background: #dcdcdc;
}

nav {
  padding: 0 10px;
  height: 70px;
  width: 100%;
  max-width: 1400px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logo {
  max-width: 40px;
  width: 100%;
  margin-right: 15px;
}

nav div {
  display: flex;
  justify-content: center;
  align-items: center;
}

.burger-btn {
  display: none;
  width: 20px;
  height: 2px;
  border-radius: 5px;
  background: #000;
  transition: all 0.4s ease-in-out;
  margin: 0 15px;
  position: relative;
  transform: rotate(180deg);
}

.burger-btn.active {
  background: transparent;
  transform: rotate(180deg);
}

.burger-btn::before,
.burger-btn::after {
  content: "";
  position: absolute;
  width: 25px;
  height: 2px;
  border-radius: 5px;
  transition: all 0.4s ease-in-out;
  background: #000;
}
.burger-btn::before {
  transform: translateY(-8px);
}
.burger-btn::after {
  transform: translateY(8px);
}
.burger-btn.active::before {
  transform: translateY(0px) rotate(-45deg);
}
.burger-btn.active::after {
  transform: translateY(0px) rotate(45deg);
}

/* sidebar css */

.sidebar {
  display: none;
}

.sidebar > div:first-child {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
}

.btn {
  padding: 10px 15px;
}

.components {
  width: 100%;
  display: flex;
  flex-flow: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.subMenu {
  padding: 0px 10px;
  background: #dcdcdc;
  width: 100%;
  min-height: 40px;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: flex-start;
}

.subData {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: flex-start;
  padding: 0 0 0 20px;
}
.subData > p {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 30px;
}
@media screen and (max-width: 768px) {
  .burger-btn {
    display: block;
  }

  nav > div {
    display: none;
  }

  .sidebar {
    display: block;
    position: fixed;
    width: 100%;
    height: 100vh;
    max-width: 425px;
    top: 70px;
    left: -100%;
    background: #fff;
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    flex-flow: column;
    padding: 0px;
    transition: all 0.4s ease-in-out;
  }

  .sidebar.active {
    left: 0;
  }
}
