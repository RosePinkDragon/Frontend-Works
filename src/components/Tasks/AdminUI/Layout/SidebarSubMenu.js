import React, { useState } from "react";

const SidebarSubMenu = ({ data }) => {
  const [isActive, setIsActive] = useState(false);

  return (
    <>
      {data && (
        <>
          <div className="subMenu">
            <p onClick={() => setIsActive(!isActive)}>{data.name}</p>
          </div>
          {isActive ? (
            <div className="subData">
              {data.options.map((option, idx) => (
                <p key={idx} onClick={() => setIsActive(!isActive)}>
                  {option}
                </p>
              ))}
            </div>
          ) : (
            ""
          )}
        </>
      )}
    </>
  );
};

export default SidebarSubMenu;
