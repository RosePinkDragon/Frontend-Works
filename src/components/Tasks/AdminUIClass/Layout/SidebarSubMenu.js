import React from "react";

class SidebarSubMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isActive: false,
    };
  }

  render() {
    return (
      <>
        {this.props.data && (
          <>
            <div className="subMenu">
              <p
                onClick={() =>
                  this.setState({ isActive: !this.state.isActive })
                }
              >
                {this.props.data.name}
              </p>
            </div>
            <div className="subData">
              {this.state.isActive
                ? this.data.options.map((option, idx) => (
                    <p
                      key={idx}
                      onClick={() =>
                        this.setState({ isActive: !this.state.isActive })
                      }
                    >
                      {option}
                    </p>
                  ))
                : ""}
            </div>
          </>
        )}
      </>
    );
  }
}

export default SidebarSubMenu;
