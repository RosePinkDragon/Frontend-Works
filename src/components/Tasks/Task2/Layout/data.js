export const sidebarData = [
  {
    name: "Admin Menu",
    options: ["All Admins", "This Admin", "None"],
  },
  {
    name: "User Menu",
    options: ["All Users", "This User", "None"],
  },
  {
    name: "Client Menu",
    options: ["All Clients", "This Client", "None"],
  },
  {
    name: "Dev Menu",
    options: ["All Devs", "This Dev", "None"],
  },
];
