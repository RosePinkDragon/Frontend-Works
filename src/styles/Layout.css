header {
  position: fixed;
  display: flex;
  top: 0;
  left: 0;
  justify-content: center;
  width: 100%;
  align-items: center;
  background: #dcdcdc;
}

nav {
  padding: 0 10px;
  height: 70px;
  width: 100%;
  max-width: 1400px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logo {
  max-width: 40px;
  width: 100%;
  margin-right: 15px;
}

nav div {
  display: flex;
  justify-content: center;
  align-items: center;
}

/* animated hamburber menu button */
.burger-btn {
  display: none;
  width: 20px;
  height: 2px;
  border-radius: 5px;
  background: #000;
  transition: all 0.4s ease-in-out;
  margin: 0 15px;
  position: relative;
  transform: rotate(180deg);
}

.burger-btn.active {
  background: transparent;
  transform: rotate(180deg);
}

.burger-btn::before,
.burger-btn::after {
  content: "";
  position: absolute;
  width: 25px;
  height: 2px;
  border-radius: 5px;
  transition: all 0.4s ease-in-out;
  background: #000;
}
.burger-btn::before {
  transform: translateY(-8px);
}
.burger-btn::after {
  transform: translateY(8px);
}
.burger-btn.active::before {
  transform: translateY(0px) rotate(-45deg);
}
.burger-btn.active::after {
  transform: translateY(0px) rotate(45deg);
}

/* sidebar css */

.sidebar {
  display: block;
  position: fixed;
  width: 100%;
  height: 100vh;
  max-width: 325px;
  top: 70px;
  left: 0;
  background: #fff;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-flow: column;
  padding: 0px;
  overflow: hidden;
  transition: all 0.4s ease-in-out;
}

.sidebar > div:first-child {
  display: none;
  justify-content: center;
  align-items: center;
  padding: 10px;
}

.components {
  width: 100%;
  display: flex;
  flex-flow: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.subMenu {
  padding: 0px 10px;
  background: #dcdcdc;
  width: 100%;
  min-height: 40px;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: flex-start;
}

.subData {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: flex-start;
  padding: 0 0 0 20px;
}
.subData > p {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 30px;
}

.bodyWrap {
  margin: 70px 0 0 350px;
}

@media screen and (max-width: 768px) {
  .burger-btn {
    display: block;
  }

  nav > div {
    display: none;
  }

  .sidebar {
    left: -120%;
  }
  .sidebar div:first-child {
    display: flex;
  }

  .sidebar.active {
    left: 0;
  }

  .bodyWrap {
    margin: 70px auto;
  }
}
