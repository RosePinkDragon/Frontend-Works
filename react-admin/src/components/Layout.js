import React, { useState } from "react";
import userLogo from "../images/userLogo.png";
import "../styles/Layout.css";
import { sidebarData } from "./data";
import SidebarSubMenu from "./SidebarSubMenu";

export default function Layout({ children }) {
  const [isActive, setIsActive] = useState(false);

  return (
    <>
      <header className="navWrap">
        <nav>
          <h1>App Name</h1>
          <div>
            <img className="logo" src={userLogo} alt="" />
            <p>Moheb Dabilkar</p>
          </div>
          <div
            className={`burger-btn ${isActive ? "active" : ""}`}
            onClick={() => setIsActive(!isActive)}
          />
        </nav>
      </header>
      <div className={`sidebar ${isActive ? "active" : ""}`}>
        <div>
          <img className="logo" src={userLogo} alt="" />
          <p>Moheb Dabilkar</p>
        </div>
        <div className="components">
          {sidebarData.map((data) => (
            <SidebarSubMenu data={data} />
          ))}
        </div>
        <div>
          <button className="btn">LogOut</button>
        </div>
      </div>
      {children}
    </>
  );
}
