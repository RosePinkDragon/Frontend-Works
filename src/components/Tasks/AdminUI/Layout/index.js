import React, { useState } from "react";
import userLogo from "images/userLogo.png";
import "styles/Layout.css";
import { sidebarData } from "./data";
import SidebarSubMenu from "./SidebarSubMenu";

export default function Layout({ children, logger }) {
  const [isActive, setIsActive] = useState(false);

  return (
    <>
      <header>
        <nav>
          <h1>App Name</h1>
          <div>
            <img className="logo" src={userLogo} alt="" />
            <p>Moheb Dabilkar</p>
          </div>
          <div
            className={`burger-btn ${isActive ? "active" : ""}`}
            onClick={() => setIsActive(!isActive)}
          />
        </nav>
      </header>
      <div className={`sidebar ${isActive ? "active" : ""}`}>
        <div className="sidebarUserInfo">
          <img className="logo" src={userLogo} alt="" />
          <p>Moheb Dabilkar</p>
        </div>
        <div className="components">
          {sidebarData.map((data, idx) => (
            <SidebarSubMenu key={idx} data={data} />
          ))}
        </div>
        <div>
          <button className="btn logOut" onClick={() => logger(false)}>
            LogOut
          </button>
        </div>
      </div>
      <div className="bodyWrap">{children}</div>
    </>
  );
}
