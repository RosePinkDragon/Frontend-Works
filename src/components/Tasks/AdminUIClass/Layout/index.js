import React from "react";
import userLogo from "images/userLogo.png";
import "styles/Layout.css";
import { sidebarData } from "utils/layoutData";
import SidebarSubMenu from "./SidebarSubMenu";
import { connect } from "react-redux";
import { logOut } from "../redux/actions/loginActions";

class Layout extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isActive: false,
    };
  }

  handleLogOut = () => {
    this.props.logOut();
  };

  render() {
    return (
      <>
        <header>
          <nav>
            <h1>App Name</h1>
            <div>
              <img className="logo" src={userLogo} alt="" />
              <p>Moheb Dabilkar</p>
            </div>
            <div
              className={`burger-btn ${this.state.isActive ? "active" : ""}`}
              onClick={() => this.setState({ isActive: !this.state.isActive })}
            />
          </nav>
        </header>
        <div className={`sidebar ${this.state.isActive ? "active" : ""}`}>
          <div className="sidebarUserInfo">
            <img className="logo" src={userLogo} alt="" />
            <p>Moheb Dabilkar</p>
          </div>
          <div className="components">
            {sidebarData.map((data, idx) => (
              <SidebarSubMenu key={idx} data={data} />
            ))}
          </div>
          <div>
            <button className="btn logOut" onClick={() => this.handleLogOut()}>
              LogOut
            </button>
          </div>
        </div>
        <div className="bodyWrap">{this.props.children}</div>
      </>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return { logged: state.logged };
};

const mapDispatchToProps = (dispatch) => {
  return {
    logOut: () => {
      dispatch(logOut());
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
