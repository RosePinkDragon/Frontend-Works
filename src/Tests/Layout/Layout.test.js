import React from "react";
import Layout from "components/Tasks/AdminUI/Layout";
// import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

const layoutChildren = "Test Children";

let wrapper;

describe("Renders the Layout with Children", () => {
  wrapper = shallow(<Layout>{layoutChildren}</Layout>);
  test("Snapshot Renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
  expect(wrapper.find(".bodyWrap").text()).toEqual(layoutChildren);
});

describe("updates state onClick", () => {
  let wrapper = mount(<Layout>{layoutChildren}</Layout>);
  wrapper.find(".burger-btn").simulate("click");
  expect(wrapper.find(".active").length).toEqual(2);
});

describe("log's out onClick", () => {
  let data = true;

  const logger = (change) => {
    data = change;
  };

  let wrapper = mount(<Layout logger={logger}>{layoutChildren}</Layout>);
  wrapper.find(".btn.logOut").simulate("click");
  expect(data).toEqual(false);
});
