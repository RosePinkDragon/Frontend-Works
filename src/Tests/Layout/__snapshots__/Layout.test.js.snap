// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Layout with Children Snapshot Renders 1`] = `
<Fragment>
  <header>
    <nav>
      <h1>
        App Name
      </h1>
      <div>
        <img
          alt=""
          className="logo"
          src="userLogo.png"
        />
        <p>
          Moheb Dabilkar
        </p>
      </div>
      <div
        className="burger-btn "
        onClick={[Function]}
      />
    </nav>
  </header>
  <div
    className="sidebar "
  >
    <div
      className="sidebarUserInfo"
    >
      <img
        alt=""
        className="logo"
        src="userLogo.png"
      />
      <p>
        Moheb Dabilkar
      </p>
    </div>
    <div
      className="components"
    >
      <SidebarSubMenu
        data={
          Object {
            "name": "Admin Menu",
            "options": Array [
              "All Admins",
              "This Admin",
              "None",
            ],
          }
        }
        key="0"
      />
      <SidebarSubMenu
        data={
          Object {
            "name": "User Menu",
            "options": Array [
              "All Users",
              "This User",
              "None",
            ],
          }
        }
        key="1"
      />
      <SidebarSubMenu
        data={
          Object {
            "name": "Client Menu",
            "options": Array [
              "All Clients",
              "This Client",
              "None",
            ],
          }
        }
        key="2"
      />
      <SidebarSubMenu
        data={
          Object {
            "name": "Dev Menu",
            "options": Array [
              "All Devs",
              "This Dev",
              "None",
            ],
          }
        }
        key="3"
      />
    </div>
    <div>
      <button
        className="btn logOut"
        onClick={[Function]}
      >
        LogOut
      </button>
    </div>
  </div>
  <div
    className="bodyWrap"
  >
    Test Children
  </div>
</Fragment>
`;
